src/util.rs

```diff
@@ -170,4 +185,18 @@ mod tests {
         assert_eq!(Vec::<PathBuf>::new(), get_expired_files(&current_dir));
         Ok(())
     }
+
+    #[test]
+    fn test_safe_join_path() {
```

Comment:
Can you add another unit test for confirming that this util function and standard `join` has the same behavior?